src/lib.rs

```diff
@@ -243,6 +246,25 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
+
+#[cfg(target_os = "macos")]
+fn get_num_physical_cpus() -> usize {
+    let mut cpus: i32 = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+
+    let sysctl_name = CString::new("hw.physicalcpu").unwrap();
```

Comment:
According to the documentation, it accepts an [ASCII string](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/sysctlbyname.3.html), which is a 7-bit only encoding - there is no difference between an ASCII string stored as [i8]s or [u8]s. Therefore, it's safe to use a `mem::transmute`